src/vcpkg/triplet.cpp

```diff
@@ -83,13 +84,33 @@ namespace vcpkg
         return Triplet::from_canonical_name(std::move(canonical_name));
     }
 
-    Triplet default_triplet(const VcpkgCmdArguments& args)
+    Triplet default_triplet(const VcpkgCmdArguments& args, bool is_manifest_mode)
     {
+        (void)is_manifest_mode;
         if (auto triplet = args.triplet.get())
         {
+            // The triplet is set by --triplet or VCPKG_DEFAULT_TRIPLET.
             return Triplet::from_canonical_name(*triplet);
         }
 #if defined(_WIN32)
+        // The triplet is not specified or is part of a string qualifier i.e. zlib:x64-windows
+        if (is_manifest_mode)
+        {
+            msg::println_warning(msgDefaultTriplet);
+        }
+        else
+        {
+            for (auto&& arg : args.command_arguments)
```

Comment:
Does this trigger on `vcpkg search` or `vcpkg create` or `vcpkg find`?

Reply:
I thought those commands do not use the default triplet?

Reply:
Then that's good -- the answer's no. I just wanted to confirm that you've checked them.

Reply:
Oh -- does this trigger on an invalid command? `vcpkg asdf zlib`?

Reply:
Yes it will trigger for that.